File: src/main/java/com/github/bjansen/mintellij/lang/MintFileType.java
package com.github.bjansen.mintellij.lang;

import com.intellij.openapi.fileTypes.LanguageFileType;
import com.intellij.openapi.util.IconLoader;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

import javax.swing.*;

public class MintFileType extends LanguageFileType {

	public static final MintFileType INSTANCE = new MintFileType();

	private MintFileType() {
		super(MintLanguage.INSTANCE);
	}

	@NotNull
	@Override
	public String getName() {
		return "Mint File";
	}

	@NotNull
	@Override
	public String getDescription() {
		return "Mint";
	}

	@NotNull
	@Override
	public String getDefaultExtension() {
		return "mint";
	}

	@Nullable
	@Override
	public Icon getIcon() {
		return IconLoader
				.getIcon("/mint/icons/logo.svg");
	}
}

File: src/main/kotlin/com/github/bjansen/mintellij/lang/MintFileType.kt
package com.github.bjansen.mintellij.lang

import com.github.bjansen.mintellij.icons.MintIcons
import com.intellij.openapi.fileTypes.LanguageFileType

object MintFileType : LanguageFileType(MintLanguage) {
	override fun getName() = "Mint File"

	override fun getDescription() = "Mint"

	override fun getDefaultExtension() = "mint"

	override fun getIcon() = MintIcons.file
}
